pubsub/gcp/pubsub/pubsub.go

```diff
@@ -175,11 +176,12 @@ func (g *GCPPubSub) Publish(ctx context.Context, req *pubsub.PublishRequest) err
 		return errors.New("component is closed")
 	}
 
-	if !g.metadata.DisableEntityManagement {
+	if !g.metadata.DisableEntityManagement && !g.topicCache[req.Topic] {
```

Comment:
Don't we want to invalidate the cache after some time so that it still checks periodically?

Reply:
Good Catch !. Any suggestions for how much time ?

Reply:
Does something like 5 hours work ?